src/python/tests/test_mlss_species_set_adaptor.py

```diff
@@ -0,0 +1,66 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of :mod:`config` module.
+
+The unit testing is divided into one test class per submodule/class found in this module, and one test method
+per public function/class method.
+
+Typical usage example::
+
+    $ pytest test_mlss_species_set_adaptor.py
+
+"""
+
+import os
+from contextlib import nullcontext as does_not_raise
+from typing import ContextManager, List
+from xml.etree import ElementTree
+
+import pytest
+from pytest import raises
+
+from ensembl.compara.config import get_species_set_by_name
+
+test_files_dir = os.path.join(os.path.dirname(__file__), "flatfiles/config/")
+
+@pytest.mark.parametrize(
+    "file, name, exp_output, expectation",
+    [
+        ("mlss_conf_simple.xml", "test1", ["danio_rerio", "gallus_gallus", "homo_sapiens", "mus_musculus", "strigamia_maritima"], does_not_raise()),
+        ("mlss_conf_simple.xml", "test3", ["", "gallus_gallus", "homo_sapiens"], does_not_raise()),
+        ("mlss_conf_simple.xml", "", ["danio_rerio", "mus_musculus"], does_not_raise()),
+        ("mlss_conf_simple.xml", "test4", [], does_not_raise()),
+        ("mlss_conf_realistic.xml", "default", ["drosophila_melanogaster", "caenorhabditis_elegans", "saccharomyces_cerevisiae"], does_not_raise()),
+        ("mlss_conf_realistic.xml", "murinae", [], does_not_raise()),
+        ("fake/path/mlss_simple.xml", "test", None, raises(FileNotFoundError, match=r"mlss_conf file not found.")),
+        ("mlss_conf_not_xml.xml", "test", None, raises(ElementTree.ParseError)),
+        ("mlss_conf_simple.xml", "test", None, raises(NameError, match=r"Species set test not found.")),
+        ("mlss_conf_simple.xml", "test2", None, raises(NameError, match=r"2 species sets named test2 found."))
```

Comment:
The 4 failure tests are ok, but you are testing the `does_not_raise` part too much since there are only 2 list comprehensions involved. I would rework this as follows: one `mlss_conf.txt` for the `ElementTree.ParseError` (so it is clear it is not XML), one `mlss_conf_dups.xml` (for instance) with the 2 species set for `RuntimeError`, and one with two correct collections (`mlss_conf.xml`?), one empty, and one with 2 genomes for the `does_not_raise`. And the `NameError` can be testes in either of the last two files.